test/functional/rpc_createmultisig.py

```diff
@@ -83,6 +84,21 @@ def check_addmultisigaddress_errors(self):
         pubs = [self.nodes[1].getaddressinfo(addr)["pubkey"] for addr in addresses]
         assert_raises_rpc_error(-5, "Bech32m multisig addresses cannot be created with legacy wallets", self.nodes[0].addmultisigaddress, 2, pubs, "", "bech32m")
 
+    def test_multisig_script_limit(self):
+        node1 = self.nodes[1]
+        pubkeys = self.pub[0:20]
+
+        self.log.info('Test legacy redeem script max size limit')
+        assert_raises_rpc_error(-8, "redeemScript exceeds size limit: 684 > 520", node1.createmultisig, 16, pubkeys, 'legacy')
+
+        self.log.info('Test valid 16-20 multisig p2sh-legacy and bech32 (no wallet)')
+        self.do_multisig(nkeys=20, nsigs=16, output_type="p2sh-segwit", wallet_multi=None)
+        self.do_multisig(nkeys=20, nsigs=16, output_type="bech32", wallet_multi=None)
+
+        self.log.info('Test invalid 16-21 multisig p2sh-legacy and bech32 (no wallet)')
+        assert_raises_rpc_error(-8, "Number of keys involved in the multisignature address creation > 20", node1.createmultisig, 16, self.pub, 'p2sh-segwit')
+        assert_raises_rpc_error(-8, "Number of keys involved in the multisignature address creation > 20", node1.createmultisig, 16, self.pub, 'bech32')
```

Comment:
```
for addr_type in ['p2sh-segwit', 'bech32']:
 assert_raises_rpc_error(-8, "Number of keys involved in the multisignature address creation > 20", node1.createmultisig, 16, self.pub, addr_type)
```

Reply:
sure, will change it if I have to retouch this code.